addons/mail/static/tests/gif_picker/gif_picker_test.js

```diff
@@ -0,0 +1,125 @@
+    await waitUntil(".o-discuss-gif", 2);
+    assert.strictEqual(
+        document.querySelector("input[placeholder='Search for a gif']").value,
+        "cry"
+    );
+});
+
+QUnit.test("Reopen gif category", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    await click(".o-discuss-gif-picker-popover .fa-arrow-left");
+    assert.containsOnce($, ".o-discuss-gif-categories");
+});
+
+QUnit.test("Make a gif favorite", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    const firstGif = document.querySelector(".o-discuss-gif .fa-star-o");
+    await click(firstGif);
+    assert.containsOnce($, ".o-discuss-gif .fa-star");
+    await click(".o-discuss-gif-picker-popover .fa-arrow-left");
+    await click(".o-discuss-gif-category:contains(Favorites)");
+    assert.containsOnce($, ".o-discuss-gif");
+});
+
+QUnit.test("Remove a gif favorite", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    await click("img[data-src='https://media.tenor.com/6uIlQAHIkNoAAAAM/cry.gif']");
+    const firstGif = document.querySelector(".o-discuss-gif .fa-star-o");
+    await click(firstGif);
+    await click(".o-discuss-gif-picker-popover .fa-arrow-left");
+    await click(".o-discuss-gif-category:contains(Favorites)");
+    await click(".o-discuss-gif .fa-star");
+    assert.containsNone($, ".o-discuss-gif");
+});
+
+QUnit.test("Chatter should not have the gif button", async (assert) => {
+    const { openFormView, pyEnv } = await start();
+    const partnerId = pyEnv["res.partner"].create({ name: "John Doe" });
+    await openFormView("res.partner", partnerId);
+    await click("button:contains(Log note)");
+    assert.containsNone($, "button[aria-label='Gifs']");
+});
+
+QUnit.test("Composer gif button should open the gif picker", async (assert) => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "" });
+    const { openDiscuss } = await start();
+    await openDiscuss(channelId);
+    await click("button[aria-label='Gifs']");
+    assert.containsOnce($, ".o-discuss-gif-picker-popover");
+});
+
+QUnit.test("Mobile: Composer gif button should open the gif picker keyboard", async (assert) => {
```

Comment:
I would avoid "Mobile: " in naming of test. Name of test should be a sentence, so it's easy to read and it tells exactly what is tested functionally.